scripts/prepare_compliance_data.py

```diff
@@ -0,0 +1,155 @@
+        shutil.copy(path.join(self.inputDirectory, "referenceset_hg37.json"),
+                    path.join(self.refsetsDirectory, "hg37.json"))
+
+        os.makedirs(self.hg37Directory)
+        for refFile in self.referenceFiles:
+            refBase = path.splitext(refFile)[0]
+            destFastaFilename = path.join(self.hg37Directory, refBase) + ".fa"
+            shutil.copy(path.join(self.inputDirectory, refBase) + ".fa",
+                        destFastaFilename)
+            pysam.tabix_compress(destFastaFilename, destFastaFilename + ".gz")
+            pysam.FastaFile(destFastaFilename + ".gz")
+            os.remove(destFastaFilename)
+            shutil.copy(path.join(self.inputDirectory, refBase) + ".json",
+                        path.join(self.hg37Directory, refBase) + ".json")
+
+        # Clean out, make and repopulate dataset directories
+        shutil.rmtree(self.datasetsDirectory, ignore_errors=True)
+        os.makedirs(self.datasetsDirectory)
+
+        for ds in self.datasets:
+            dsdir = path.join(self.datasetsDirectory, ds)
+            os.makedirs(dsdir)
+
+            # Reads
+            print("Converting reads...", file=sys.stderr)
+            dsReadsdir = path.join(dsdir, "reads")
+            os.makedirs(dsReadsdir)
+            for readFile in self.datasetReads[ds]:
+                destFile = path.join(dsReadsdir,
+                                     readFile.split('_')[1].split('.')[0]) \
+                           + ".bam"
+                readSrc = pysam.AlignmentFile(path.join(self.inputDirectory,
+                                                        readFile), "r")
+                readDest = pysam.AlignmentFile(destFile, "wb",
+                                               header=readSrc.header)
+                destFilePath = readDest.filename
+
+                for readData in readSrc:
+                    readDest.write(readData)
+                readDest.close()
+                readSrc.close()
+                pysam.index(destFilePath)
+
+            # Variants
+            print("Converting variants...", file=sys.stderr)
+            dsVariantsdir = path.join(dsdir, "variants")
+            os.makedirs(dsVariantsdir)
+            for vgroup in self.datasetVariants[ds].keys():
+                vgroupdir = path.join(dsVariantsdir, vgroup)
+                os.makedirs(vgroupdir)
+                for variantFile in self.datasetVariants[ds][vgroup]:
+                    destFile = path.join(vgroupdir, variantFile.split('_')[2])
+                    shutil.copy(path.join(self.inputDirectory, variantFile),
+                                destFile)
+                    # Pysam's tabix_index automatically compresses the file
+                    # in place, creates a tabix index.
+                    pysam.tabix_index(destFile, preset="vcf")
+
+    def cleanup(self):
+        print("done converting compliance data.", file=sys.stderr)
+
+
+def main():
```

Comment:
I like putting the utils.Timed decorator on the main function of scripts, just to see how long they take to execute; it's often helpful.